### PR TITLE
Create an itc tile

### PR DESCRIPTION
**Pull request by cquiroz**
This PR sets up an empty ITC tile

Besides it collects the parameters and sends a query to the chart endpoint of the itc getting the data back into the tile

### CODE REVIEW

File: common-graphql/src/main/resources/graphql/schemas/ITC.graphql [hunk 1]
```diff
@@ -4718,9 +4746,32 @@ type Result {
...
+  serverVersion: String!
 
   # Search results
   results: [Result!]!
 }
 
+type ItcSeries {
+  title: String!
+
+  # Data type: alternatives SignalData, BackgroundData, SingleS2NData, FinalS2NData, PixSigData, PixBackData
```
Comment (rpiaggio) [inline common-graphql/src/main/resources/graphql/schemas/ITC.graphql:4759]: Can this be an `enum`?

File: explore/src/main/scala/explore/itc/ItcGraphPanel.scala [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+              OptionT.fromOption(
+                props.scienceData.flatMap(r => NonEmptyList.fromList(r.itcTargets))
+              )
+            mode        <- OptionT.fromOption(props.instrumentRow)
+          yield uuid
+            .map(u =>
+              queryId.setState(uuid).to[IO] *> props.ctx.worker
+                .postWorkerMessage(ItcGraphQuery(u, w, sn, constraints, t, mode))
+            )
+            .orEmpty).getOrElse(IO.unit).flatten
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/itc/ItcGraphPanel.scala:134]: `orEmpty`

Comment (rpiaggio) [reply explore/src/main/scala/explore/itc/ItcGraphPanel.scala:134]: oh wait, is that invoked on the `OptionT`? IDK if `orEmpty` works there

File: model/shared/src/main/scala/explore/model/ScienceModeBasic.scala [hunk 3]
```diff
@@ -16,27 +17,18 @@ import lucuma.core.enums.GmosSouthGrating
...
-      case (a: GmosSouthLongSlit, b: GmosSouthLongSlit) => a === b
-      case _                                            => false
-    }
-
   final case class GmosNorthLongSlit(
     grating: GmosNorthGrating,
     filter:  Option[GmosNorthFilter],
     fpu:     GmosNorthFpu
   ) extends ScienceModeBasic
+      derives Eq
```
Comment (rpiaggio) [inline model/shared/src/main/scala/explore/model/ScienceModeBasic.scala:28]: I'm wondering if this `derives Eq` is necessary or the one in the trait derives anyway and kicks in when we compare subtypes.

File: workers/src/main/scala/workers/CacheIDBWorker.scala [hunk 4]
```diff
@@ -53,7 +54,7 @@ object CacheIDBWorker extends CatalogCache with EventPicklers with AsyncToIO {
   def runWorker(): Unit = run.handleError(t => t.printStackTrace()).unsafeRunAndForget()
 
   def setupLogger[F[_]: Sync]: F[Logger[F]] = Sync[F].delay {
-    LogLevelLogger.setLevel(LogLevelDesc.INFO)
+    LogLevelLogger.setLevel(LogLevelDesc.DEBUG)
```
Comment (rpiaggio) [inline workers/src/main/scala/workers/CacheIDBWorker.scala:57]: Log level

**Review by rpiaggio — APPROVED** [review]
LGTM, thank you for all the migration to Scala 3

File: model/shared/src/main/scala/explore/model/ScienceModeBasic.scala (hunk 3 above)
Comment (cquiroz) [reply model/shared/src/main/scala/explore/model/ScienceModeBasic.scala:28]: Good question. I guess it depends on whether you ever call `===` on the subtype

File: common-graphql/src/main/resources/graphql/schemas/ITC.graphql (hunk 1 above)
Comment (cquiroz) [reply common-graphql/src/main/resources/graphql/schemas/ITC.graphql:4759]: as in graphql enum?
I need to go back to the schema and will review that

File: model/shared/src/main/scala/explore/model/ScienceModeBasic.scala (hunk 3 above)
Comment (cquiroz) [reply model/shared/src/main/scala/explore/model/ScienceModeBasic.scala:28]: They are used to define `Reusaibility` somewhere

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 5]
```diff
@@ -66,13 +66,16 @@ final case class ObsTabContents(
...
+enum ObsTabTilesIds:
+  case NotesId, TargetId, PlotId, ConstraintsId, ConfigurationId, ItcId
+
+  def id: NonEmptyString = this match
+    case NotesId         => "notes".refined
+    case TargetId        => "target".refined
+    case PlotId          => "elevationPlot".refined
+    case ConstraintsId   => "constraints".refined
+    case ConfigurationId => "configuration".refined
+    case ItcId           => "itc".refined
```
Comment (toddburnside) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:78]: An alternative would be to use `enum ObsTabTilesIds(val id: NonEmptyString)` and skip the match here.

**Review by toddburnside — APPROVED** [review]
LGTM

File: workers/src/main/scala/workers/itc/ITCGraphRequests.scala [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+import org.typelevel.log4cats.Logger
+import queries.common.ITCQueriesGQL._
+import queries.schemas.ITC
+import queries.schemas.itc.implicits.*
+
+import java.util.UUID
+import scala.concurrent.duration._
+//
+// Find the magnitude closest to the requested wavelength
+def selectedBrightness(
```
Comment (cquiroz) [inline workers/src/main/scala/workers/itc/ITCGraphRequests.scala:39]: I do like these "free" functions

**Pull request merged**